### PR TITLE
Add CLI

### PR DESCRIPTION
**Pull request by doronz88**
I'm also considering either scraping TheIphoneWiki or including a JSON file to store all encryption keys.
This is in continuation to #4

### CODE REVIEW

File: pyimg4/parser.py [hunk 1]
```diff
@@ -728,6 +728,10 @@ def _create_complzss_header(self) -> bytes:
 
         return bytes(header)
 
+    @property
+    def data(self) -> bytes:
```
Comment (m1stadev) [inline pyimg4/parser.py:732]: `pyimg4.IM4PData` subclasses `pyimg4._PyIMG4`, which provides the `_PyIMG4.output()` function, which does the same thing, so this is unnecessary.

**Review by m1stadev — CHANGES_REQUESTED** [review]

File: pyimg4/parser.py (hunk 1 above)
Comment (doronz88) [reply pyimg4/parser.py]: 👍

**Comment by m1stadev** [comment]
I'm gonna make the CLI more feature complete before merging, so it may be a while.

**Comment by doronz88** [comment]
You mean to the inclusion for both serializing and deserializing of all permutations? I can add this also.

**Comment by doronz88** [comment]
Something like:
```shell
pyimg4 im4p create FILENAME DATA [--compression ..] [--key. ..] [--iv ...]
```

**Comment by m1stadev** [comment]
> You mean to the inclusion for both serializing and deserializing of all permutations? I can add this also.

Yeah. I'll probably work on adding some commands tonight.

File: pyimg4/__main__.py [hunk 2]
```diff
@@ -14,6 +14,47 @@ def cli():
...
 
+@cli.group()
+def im4m() -> None:
+    '''Image4 manifest commands'''
+
+    pass
+
+
+@im4m.command('info')
+@click.option(
```
Comment (doronz88) [inline pyimg4/__main__.py:25]: since this is always required and only one parameter its better to be an argument instead of an option

Comment (m1stadev) [reply pyimg4/__main__.py:25]: I'd prefer for it to stay as an option for parity with other commands that do require input as an option

**Comment by m1stadev** [comment]
I think the CLI should be ready to merge now. @doronz88 do you think there's any other features that should be included?

**Comment by doronz88** [comment]
Just for the git history sake, I think its better to rebase it now before merging

**Review by m1stadev — APPROVED** [review]

**Pull request merged**